src/gui/qgsmessagelogviewer.cpp

```diff
@@ -41,6 +42,47 @@ QgsMessageLogViewer::QgsMessageLogViewer( QWidget *parent, Qt::WindowFlags fl )
            this, static_cast<void ( QgsMessageLogViewer::* )( const QString &, const QString &, Qgis::MessageLevel )>( &QgsMessageLogViewer::logMessage ) );
 
   connect( tabWidget, &QTabWidget::tabCloseRequested, this, &QgsMessageLogViewer::closeTab );
+
+  tabWidget->tabBar()->setContextMenuPolicy( Qt::CustomContextMenu );
+  connect( tabWidget->tabBar(), &QWidget::customContextMenuRequested, this, &QgsMessageLogViewer::showContextMenuForTabBar );
+}
+
+void QgsMessageLogViewer::showContextMenuForTabBar( QPoint point )
+{
+  if ( point.isNull() )
+  {
+    return;
+  }
+
+  int tabIndex = tabWidget->tabBar()->tabAt( point );
+  QMenu *menu = new QMenu( this );
```

Comment:
Menu is leaked here. Well, not really leaked (since it's cleaned up at log destruction), but it's kept around for the lifetime of QGIS.

I'd suggest making mMenu a member here, and creating it once in the widget constructor. Then replace this line with

    mMenu->clear();

Reply:
@nyalldawson thanks! So also both actions should be members, just connected and disconnected to the lambda every time, shouldn't they? Now I see I mistakenly named them like mActionCloseTab, despite they are not members (yet).

Reply:
Actually I'd leave them as constructed on demand like you have here, and only make the menu a member. If you clear it just before showing it will delete the existing actions, so at most only the menu+ 2 actions will exist...

Reply:
Great, now should be ok.